### PR TITLE
remove password step from FB signin on public library/profile page

### PR DESCRIPTION
**Pull request by ahsu1230**
Go to a public library/profile page
click on Follow or Sign In
click on Continue with Facebook

After authenticating, you normally would be brought to a finalize modal (You're almost there! Enter in a password phase)
We want to skip this step and just go straight to "Thanks for Registering" install extension modal.

@andrewconner

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/signup/signupService.js [hunk 1]
```diff
@@ -166,9 +166,13 @@ angular.module('kifi')
...
             } else if (resp.code && resp.code === 'continue_signup') {
               // todo: handle case when emails match, so backend auto-logs in user - andrew
-              modalService.close();
+
               $scope.userData.method = 'social';
-              registerFinalizeModal();
+
+              // skip password setting (set randomly)
+              $scope.userData.password = Math.random().toString(36).substring(8);
+              $scope.registerFinalizeSubmit();
```
Comment (andrewconner) [inline kifi-backend/modules/shoebox/angular/src/signup/signupService.js:174]: Shouldn't the server handle this? ie, user auths with Facebook, call server to register, and when it responds, they're finished.

**Comment by andrewconner** [comment]
Since you're leaving them on the page for this, are you doing the same for twitter?

**Comment by ahsu1230** [comment]
did we agree that we would use `signup/twitter` for authentication? and use cookies to redirect them back?

**Comment by andrewconner** [comment]
If you want to do it that way for twitter, why not just do facebook the same way? Will be much easier to maintain in the future.

**Comment by ahsu1230** [comment]
Ok that sounds good. The facebook pop-up gets blocked by ad-blockers anyway lol.

**Pull request closed without merging**